typescript/sdk/src/token/CosmosNativeWarpRouteReader.ts

```diff
@@ -0,0 +1,189 @@
+   * @returns The derived token type, which can be one of: collateralVault, collateral, native, or synthetic.
+   */
+  async deriveTokenType(warpRouteAddress: Address): Promise<TokenType> {
+    const { token } = await this.cosmosProviderOrSigner.query.warp.Token({
+      id: warpRouteAddress,
+    });
+
+    if (!token) {
+      throw new Error(`Failed to find token for address ${warpRouteAddress}`);
+    }
+
+    switch (token.token_type) {
+      case warpTypes.HypTokenType.HYP_TOKEN_TYPE_COLLATERAL:
+        return TokenType.collateral;
+      case warpTypes.HypTokenType.HYP_TOKEN_TYPE_SYNTHETIC:
+        return TokenType.synthetic;
+      default:
+        throw new Error(
+          `Failed to determine token type for address ${warpRouteAddress}`,
+        );
+    }
+  }
+
+  /**
+   * Fetches the base metadata for a Warp Route contract.
+   *
+   * @param routerAddress - The address of the Warp Route contract.
+   * @returns The base metadata for the Warp Route contract, including the mailbox, owner, hook, and ism.
+   */
+  async fetchMailboxClientConfig(
+    routerAddress: Address,
+  ): Promise<MailboxClientConfig> {
+    const { token } = await this.cosmosProviderOrSigner.query.warp.Token({
+      id: routerAddress,
+    });
+
+    assert(token, `Failed to find token for address ${routerAddress}`);
+
+    const config: MailboxClientConfig = {
+      mailbox: token.origin_mailbox,
+      owner: token.owner,
+    };
+
+    if (token.ism_id) {
+      const derivedIsm = await this.ismReader.deriveIsmConfig(token.ism_id);
+      config.interchainSecurityModule = derivedIsm;
+    }
+
+    return config;
+  }
+
+  /**
+   * Fetches the metadata for a token address.
+   *
+   * @param warpRouteAddress - The address of the token.
+   * @returns A partial ERC20 metadata object containing the token name, symbol, total supply, and decimals.
+   * Throws if unsupported token type
+   */
+  async fetchTokenConfig(
+    type: TokenType,
+    warpRouteAddress: Address,
+  ): Promise<HypTokenConfig> {
+    return {
+      type,
+      token: warpRouteAddress,
+    } as HypTokenConfig;
+  }
+
+  async fetchRemoteRouters(warpRouteAddress: Address): Promise<RemoteRouters> {
+    const { remote_routers } =
+      await this.cosmosProviderOrSigner.query.warp.RemoteRouters({
+        id: warpRouteAddress,
+      });
+
+    const routers: Record<string, { address: string }> = {};
+    for (const router of remote_routers) {
+      routers[router.receiver_domain] = {
+        address: router.receiver_contract,
+      };
+    }
+
+    return RemoteRoutersSchema.parse(routers);
+  }
+
+  async fetchProxyAdminConfig(
+    tokenAddress: Address,
+  ): Promise<DeployedOwnableConfig> {
+    const { token } = await this.cosmosProviderOrSigner.query.warp.Token({
+      id: tokenAddress,
+    });
+
+    return {
+      address: tokenAddress,
+      owner: token?.owner ?? '',
+    };
+  }
```

Comment:
considering that cosmos does not have a proxy admin i think you can delete this and leave the `proxyAdmin` as undefined to aovid giving the idea that there is a proxy admin when there is none